File: frontend/src/types/history.ts
import { Decision } from './detection';

export interface HistoryRecord {
  request_id: string;
  prompt: string;
  generated_response: string;
  risk_score: number;
  decision: Decision;
  created_at: string;
}

export interface HistoryQueryParams {
  limit?: number;
  offset?: number;
  decision?: Decision;
}

File: backend/app/schemas/history.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import Decision


class HistoryRecord(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    request_id: str
    prompt: str
    generated_response: str
    risk_score: int
    decision: Decision
    created_at: datetime
